modules/samtools_dedup.cfmod.pl

```diff
@@ -0,0 +1,166 @@
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+# Get Options
+my $required_cores;
+my $required_mem;
+my $required_modules;
+my $help;
+my $result = GetOptions ("cores=i" => \$required_cores, "mem=s" => \$required_mem, "modules" => \$required_modules, "help" => \$help);
+
+# QSUB SETUP
+# --cores i = offered cores. Return number of required cores.
+if($required_cores){
+	print CF::Helpers::allocate_cores($required_cores, 2, 2);
+	exit;
+}
+# --mem. Return the required memory allocation.
+if($required_mem){
+	print CF::Helpers::allocate_memory($required_mem, '5G', '5G');
+	exit;
+}
+# --modules. Return csv names of any modules which should be loaded.
+if($required_modules){
+	print 'samtools';
+	exit;
+}
+# --help. Print help.
+if($help){
+	print "".("-"x15)."\n Samtools dedup Module\n".("-"x15)."\n
+This module uses the samtools package to mark and remove duplicated
+sequences from mapped BAM files.  The process is actually 3 steps
+
+1)samtools sort
+2)samtools rmdup
+3)samtools sort -n
+
+The output will be deduplicated BAM files sorted by sequence name.
+\n\n";
+	exit;
+}
+
+# MODULE
+my $timestart = time;
+
+# Read in the input files from the run file
+my ($files, $runfile, $job_id, $prev_job_id, $cores, $mem, $parameters, $config_ref) = CF::Helpers::load_runfile_params(@ARGV);
+my %config = %$config_ref;
+
+
+if(!defined($cores) || $cores < 1){
+	$cores = 1;
+}
+
+open (RUN,'>>',$runfile) or die "Can't write to $runfile: $!";
+
+
+# Go through each file and run Tophat
+my $output_dir;
+foreach my $file (@$files){
+				
+		my $pos_sorted_fn = $file;
+		$pos_sorted_fn =~ s/.bam$//i;
+
+		my $dedup_fn = $pos_sorted_fn."_raw_dedup.bam";
+		my $dedup_resorted_fn = $pos_sorted_fn."_dedup";# Samtools automatically adds the bam in this case.
+		$pos_sorted_fn .= '_sorted'; # Samtools automatically adds the bam in this case.
+
+		# For the rmdup we need to specify whether the file is single or paired end to get the deduplication
+		# right.  We can't rely on the single/paired flag from clusterflow since the input is a bam file, so
```

Comment:
What's wrong with the [`CF::Helpers::is_bam_paired_end`](https://github.com/ewels/clusterflow/blob/master/source/CF/Helpers.pm#L393-L433) function? It looks very similar to your `test_paired_end` function here, so I'll just use that and remove the module specific code.